akka-http-core/src/main/scala/akka/http/impl/engine/parsing/HttpMessageParser.scala

```diff
@@ -96,10 +96,13 @@ private[http] trait HttpMessageParser[Output >: MessageOutput <: ParserOutput] {
         ele
     }
 
-  protected final def shouldComplete(): Boolean = {
+  protected final def shouldComplete(callerDescriptor: String): Boolean = {
     completionHandling() match {
-      case Some(x) => emit(x)
-      case None    => // nothing to do
+      case Some(x) => {
+        debugMessage(s"emitting error from $callerDescriptor with result $result")
```

Comment:
This will tell us if the ESE comes from the request or the response path.
I believe that it will be clear from other logs that I write from `emit()` below, but this is quicker to spot.